File: ruby/lib/pdf_extractor.rb
# frozen_string_literal: true

require_relative 'pdf_extractor/version'
require_relative 'pdf_extractor/extractor'
require_relative 'pdf_extractor/command_runner'
require_relative 'pdf_extractor/markdown_converter'
require_relative 'pdf_extractor/cli'

# Main module for PDF page extraction functionality
module PdfExtractor
  class Error < StandardError; end
  class CommandError < Error; end
  class FileNotFoundError < Error; end
end

File: elixir/lib/pdf_extractor.ex
defmodule PdfExtractor do
  @moduledoc """
  PDF page extractor for Elixir.

  This module provides functionality to extract specific pages from PDF files
  and optionally prepend Markdown content as styled introductions.
  """

  @doc """
  Extract pages from a PDF file based on configuration.
  """
  defdelegate extract_pages(input_pdf, output_pdf, yaml_path, markdown_path \\ nil),
    to: PdfExtractor.Core
end
